src/components/swapv2/LimitOrder/DeltaRate.tsx

```diff
@@ -8,45 +8,62 @@ import useTheme from 'hooks/useTheme'
 
 import { RateInfo } from './type'
 
-const DeltaRate = ({
+export function useGetDeltaRateLimitOrder({
   marketPrice,
   rateInfo,
-  symbolIn,
 }: {
   marketPrice: Price<Currency, Currency> | undefined
   rateInfo: RateInfo
-  symbolIn: string
-}) => {
-  const theme = useTheme()
-  let percent: number | string = ''
+}) {
+  let percent: number | string = '',
+    deltaText = ''
   try {
     if (marketPrice && rateInfo.rate && rateInfo.invertRate) {
       const { rate, invert, invertRate } = rateInfo
       const ourRate = Number(invert ? invertRate : rate)
       const marketRate = Number(invert ? marketPrice.invert().toFixed(10) : marketPrice.toFixed(10))
       percent = ((ourRate - marketRate) / marketRate) * 100
+      if (invert) percent = -percent
+      const delta = Number(percent)
+      const sign = delta > 0 ? '+' : ''
+      deltaText = `${Math.abs(delta) > 100 ? '>100' : `${sign}${delta.toFixed(2)}`}%`
     }
   } catch (error) {
     console.log(error)
   }
 
-  const delta = Number(percent)
-  const sign = delta > 0 ? '+' : ''
-  const deltaText = `${Math.abs(delta) > 100 ? '>100' : `${sign}${delta.toFixed(2)}`}%`
-  const color = delta > 0 ? theme.apr : theme.red
-  console.log(delta)
+  const percentText = Math.abs(Number(percent)) > 0.009 ? deltaText : ''
+  return {
+    percent: percentText,
+    profit: percentText && Number(percent) > 0,
+  }
+}
+
+const DeltaRate = ({
+  marketPrice,
+  rateInfo,
+  symbolIn,
+}: {
+  marketPrice: Price<Currency, Currency> | undefined
+  rateInfo: RateInfo
+  symbolIn: string
+}) => {
+  const theme = useTheme()
+
+  const { percent, profit } = useGetDeltaRateLimitOrder({ marketPrice, rateInfo })
+  const color = profit ? theme.apr : theme.red
 
   return (
     <Label style={{ display: 'flex', alignItems: 'center', whiteSpace: 'nowrap' }}>
       <Trans>Sell {symbolIn} at rate</Trans>
-      {Math.abs(delta) > 0.009 && percent ? (
+      {percent ? (
         <>
           <Text as="span" color={color}>
-            &nbsp;{deltaText}
+            &nbsp;{percent}
           </Text>
           <InfoHelper
             color={color}
-            text={t`Your selected price is ${deltaText} ${delta > 0 ? `above` : `below`} the current market price.`}
+            text={t`Your selected price is ${percent} ${profit ? t`above` : t`below`} the current market price.`}
```

Comment:
This is still not correct too 😂 

```suggestion
            text={profit ? t`Your selected price is ${percent} above the current market price.` :  `Your selected price is ${percent} below the current market price.`}
```